catkin_home/src/navigation/src/scripts/navServer.py

```diff
@@ -0,0 +1,135 @@
+
+    def searchGoal(self, goal):
+
+       # Load data
+        i=""
+        with open('src/navigation/src/data/mock_locations.json') as x:
+            goals = json.load(x)
+
+        for pos in goals:
+            # goal.target_location -> the location sent by the main_engine (specify the type on nav_action/navServ.action)
+            if(pos == goal.target_location): 
+                i  = goals[pos]["location"]
+
+       
+        if(i == ""): return False
+
+        self.x = i["x"]
+        self.y = i["y"]
+        self.z = i["z"]
+        self.qx = i["qx"]
+        self.qy = i["qy"]
+        self.qz = i["qz"]
+        self.qw = i["qw"]
+
+        return True
+      
+    def execute_cb(self, goal):
+        # Helper variables
+        r = rospy.Rate(1)
+        success = True
+        g = rospy.Publisher('goal', MoveBaseGoal, queue_size=10)
+        
+        # Start executing the action
+        # self._feedback.status = True
+        rospy.loginfo("Looking for the goal")
+        isValid = False
+        print(type(goal.target_location))
+        if(type(goal.target_location) == str):
+            isValid = self.searchGoal(goal)
+        #else:
+            #go to object action
+        if self._as.is_preempt_requested():
+
+            rospy.loginfo('%s: Preempted' % self._action_name)
+            self._as.set_preempted()
+            success = False
+            
+        
+        # Valid if the given location is in the known locations.
+        if isValid == True:
+            rospy.loginfo("Goal found!")
+
+            # Look if the move_base node is up
+            move_base_client = actionlib.SimpleActionClient('move_base',MoveBaseAction)
+            rospy.loginfo("Waiting for server response")
+
+            #move_base_client.wait_for_server()
+            #rospy.loginfo("Server found")
+
+            # Creating the goal message (PoseStamped)
+            target = MoveBaseGoal()
+            target.target_pose.header.frame_id = "goal"
+            target.target_pose.header.stamp = rospy.Time.now()
+
+            # xyz positions
+            target.target_pose.pose.position.x = self.x
+            target.target_pose.pose.position.y = self.y
+            target.target_pose.pose.position.z = self.z
+            
+            # Quaternion positions
+            target.target_pose.pose.orientation.x = self.qx
+            target.target_pose.pose.orientation.y = self.qy
+            target.target_pose.pose.orientation.z = self.qz
+            target.target_pose.pose.orientation.w = self.qw
+
+            while True:
+                g.publish(target)
+                rospy.sleep(1)
+
+            rospy.loginfo("Sending Goal to move base node")
+            #move_base_client.send_goal(target)
+            #rospy.loginfo("Goal sent!")
+            #client.wait_for_result()
+            #move_base_result = move_base_client.get_result() 
+            while (move_base_result.result != 3 or move_base_result != 1):
```

Comment:
shouldn't it be move_base_result.result != 1 ?

Reply:
This a re just default mock values, we will know the exact values of move_base_result one we make test with that node.

Reply:
Oh ok, so for now can I connect it with the main engine client as it is? Or do we need to modify something before merging with develop to achieve this?

Reply:
The only thing that is missing is the action for every time the main_engine sends coordinates instead of a string. This action is in progress of being developed.

Reply:
Ok, that can be implemented with another pull request, for now integrating main_engine + nav is key to move forward.

Reply:
Perfect!